### PR TITLE
matrix-appservice-discord: 0.5.2 -> 1.0.0

### PR DESCRIPTION
**Pull request by pacien**
###### Motivation for this change

This updates the matrix-appservice-discord package to the latest available version.
Upstream now uses Yarn instead of NPM, so we now use yarn2nix instead of node2nix.

Due to Discord API changes, the previous version will cease to work after 2020-11-07.
Migration to this newer version being mandatory, this should be backported to the release-20.09 branch as well.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-appservice-discord/default.nix [hunk 1]
```diff
@@ -24,6 +36,10 @@ in nodePackages."matrix-appservice-discord-git+https://github.com/Half-Shot/matr
     done
   '';
 
-  # other metadata generated and inherited from ./node-package.nix
-  meta.maintainers = with stdenv.lib.maintainers; [ pacien ];
+  meta = {
+    description = "A bridge between Matrix and Discord.";
+    homepage = "https://github.com/Half-Shot/matrix-appservice-discord";
+    license = lib.licenses.asl20;
```
Comment (SuperSandro2000) [inline pkgs/servers/matrix-appservice-discord/default.nix:42]: Platforms are missing.

Comment (pacien) [reply pkgs/servers/matrix-appservice-discord/default.nix:42]: Added and amended.

**Review by D4ndellion — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>matrix-appservice-discord</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102973` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>matrix-appservice-discord</li>
  </ul>
</details>

```
    version = CLTVersion()  # macOS Catalina returns 11.0.0.0.1.1567737322
  File "/nix/store/pjacii8rznr3igdv433a4na0g2y3nxxf-nodejs-14.15.3/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1320, in CLTVersion
    output = GetStdout(['/usr/sbin/pkgutil', '--pkg-info', key])
  File "/nix/store/pjacii8rznr3igdv433a4na0g2y3nxxf-nodejs-14.15.3/lib/node_modules/npm/node_modules/node-gyp/gyp/pylib/gyp/xcode_emulation.py", line 1342, in GetStdout
    job = subprocess.Popen(cmdlist, stdout=subprocess.PIPE)
  File "/nix/store/60ficlmwyiwvkmyzarjz2p8xfxpyk5if-python-2.7.18/lib/python2.7/subprocess.py", line 394, in __init__
    errread, errwrite)
  File "/nix/store/60ficlmwyiwvkmyzarjz2p8xfxpyk5if-python-2.7.18/lib/python2.7/subprocess.py", line 1047, in _execute_child
    raise child_exception
OSError: [Errno 1] Operation not permitted
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (/nix/store/pjacii8rznr3igdv433a4na0g2y3nxxf-nodejs-14.15.3/lib/node_modules/npm/node_modules/node-gyp/lib/configure.js:351:16)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
gyp ERR! System Darwin 19.6.0
gyp ERR! command "/nix/store/pjacii8rznr3igdv433a4na0g2y3nxxf-nodejs-14.15.3/bin/node" "/nix/store/pjacii8rznr3igdv433a4na0g2y3nxxf-nodejs-14.15.3/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--release"
gyp ERR! cwd /private/tmp/nix-build-matrix-appservice-discord-modules-1.0.0.drv-0/node_modules/better-sqlite3
gyp ERR! node -v v14.15.3
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! better-sqlite3@7.1.1 build-release: `node-gyp rebuild --release`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the better-sqlite3@7.1.1 build-release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /private/tmp/nix-build-matrix-appservice-discord-modules-1.0.0.drv-0/yarn_home/.npm/_logs/2020-12-27T11_43_21_878Z-debug.log

builder for '/nix/store/x0dixcfhpbmb27kzlh2bw5dk6i0lk8fs-matrix-appservice-discord-modules-1.0.0.drv' failed with exit code 1
cannot build derivation '/nix/store/ywdzvyg0m9j8ssvz2xy70h0kypvj9q7d-matrix-appservice-discord-1.0.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/ywdzvyg0m9j8ssvz2xy70h0kypvj9q7d-matrix-appservice-discord-1.0.0.drv' failed
```

**Comment by pacien** [comment]
@SuperSandro2000 Thanks for testing; I updated the package metadata to exclude building on Darwin.

**Comment by erictapen** [comment]
Backport in ec9cb413bacaf52d9d1fcf267d162b4d48c0b904.

**Pull request merged**